src/sap/fhir/model/r4/FHIRUtils.js

```diff
@@ -497,14 +503,61 @@ sap.ui.define([
 		} else if (oFilter.sOperator === FHIRFilterOperator.BT) {
 			var oValue2 = FHIRFilterOperatorUtils.getFilterValue(oFilter.oValue2);
 			mParameters[sPath + sFhirSearchModifier] = ["ge" + oValue1, "le" + oValue2];
-		} else if (bLogicalConnection && !bLogicalOperator){
-			if (mParameters[sPath + sFhirSearchModifier]){
+		} else if (bLogicalConnection && !bLogicalOperator) {
+			if (mParameters[sPath + sFhirSearchModifier]) {
 				mParameters[sPath + sFhirSearchModifier].push(vValue);
 			} else {
-				mParameters[sPath + sFhirSearchModifier] = [ vValue ];
+				mParameters[sPath + sFhirSearchModifier] = [vValue];
 			}
 		} else {
-			mParameters[sPath + sFhirSearchModifier] =  vValue;
+			mParameters[sPath + sFhirSearchModifier] = vValue;
+		}
+	};
+
+	/**
+	 * Creates a complex filter
+	 *
+	 * @param {object} oFilter The filter which should be added to the parameters
```

Comment:
it is only an object? It is a valid fhirfilter value, so we can update the type to that one

Reply:
done